halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/model/v1/security/OAuth2.java

```diff
@@ -41,15 +42,96 @@ public NodeIterator getChildren() {
     return NodeIteratorFactory.makeEmptyIterator();
   }
 
-  private boolean enabled;
+  @Override
+  public Method getMethod() {
+    return Method.OAuth2;
+  }
+
   private Client client = new Client();
-  private UserInfoRequirements userInfoRequirements = new UserInfoRequirements();
-  private Provider provider = null;
+  private UserInfoRequirements userInfoRequirements = null; // TODO(lwander) ensure that when hd is not set, this field is null
+  private Resource resource = new Resource();
+  private UserInfoMapping userInfoMapping = new UserInfoMapping();
+  private Provider provider;
+
+  public void setProvider(Provider provider) {
+    this.provider = provider;
+
+    if (provider == null) {
+      return;
+    }
+
+    Client newClient = new Client()
+        .setClientId(client.getClientId())
+        .setClientSecret(client.getClientSecret());
+    Resource newResource = new Resource();
+    UserInfoMapping newUserInfoMapping = new UserInfoMapping();
+
+    switch (provider) {
+      case GOOGLE:
+        newClient.setAccessTokenUri("https://www.googleapis.com/oauth2/v4/token");
+        newClient.setUserAuthorizationUri("https://accounts.google.com/o/oauth2/v2/auth");
+        newClient.setScope("profile email");
+
+        newResource.setUserInfoUri("https://www.googleapis.com/oauth2/v3/userinfo");
+
+        newUserInfoMapping.setEmail("email");
+        newUserInfoMapping.setFirstName("given_name");
+        newUserInfoMapping.setLastName("family_name");
+        break;
+      case GITHUB:
+        newClient.setAccessTokenUri("https://github.com/login/oauth/access_token");
+        newClient.setUserAuthorizationUri("https://github.com/login/oauth/authorize");
+        newClient.setScope("user:email");
+
+        newResource.setUserInfoUri("https://api.github.com/user");
+
+        newUserInfoMapping.setEmail("email");
+        newUserInfoMapping.setFirstName("");
+        newUserInfoMapping.setLastName("name");
+        newUserInfoMapping.setUsername("login");
+        break;
+      case AZURE:
+        newClient.setAccessTokenUri("https://login.microsoftonline.com/${azureTenantId}/oauth2/token");
+        newClient.setUserAuthorizationUri("https://login.microsoftonline.com/${azureTenantId}/oauth2/authorize?resource=https://graph.windows.net");
+        newClient.setScope("profile");
+        newClient.setClientAuthenticationScheme("query");
+
+        newResource.setUserInfoUri("https://graph.windows.net/me?api-version=1.6");
+
+        newUserInfoMapping.setEmail("userPrincipalName");
+        newUserInfoMapping.setFirstName("givenName");
+        newUserInfoMapping.setLastName("surname");
+        break;
+      default:
```

Comment:
Yup they have two options:

1. (easy) leave `provider == null` and then set each flag by hand using the CLI/halconfig.
2. (not-as-easy) extend halyard with a new oauth2 provider type.